pkg/cmd/cli/restic/server.go

```diff
@@ -207,18 +206,25 @@ func (s *resticServer) run() {
 		s.logger.Fatalf("Failed to create credentials file store: %v", err)
 	}
 
-	backupController := controller.NewPodVolumeBackupController(
-		s.logger,
-		s.veleroInformerFactory.Velero().V1().PodVolumeBackups(),
-		s.veleroClient.VeleroV1(),
-		s.podInformer,
-		s.kubeInformerFactory.Core().V1().PersistentVolumeClaims(),
-		s.kubeInformerFactory.Core().V1().PersistentVolumes(),
-		s.metrics,
-		s.mgr.GetClient(),
-		os.Getenv("NODE_NAME"),
-		credentialFileStore,
-	)
+	pvbReconciler := controller.PodVolumeBackupReconciler{
+		Scheme:         s.mgr.GetScheme(),
+		Client:         s.mgr.GetClient(),
+		Ctx:            s.ctx,
+		Clock:          clock.RealClock{},
+		Metrics:        s.metrics,
+		CredsFileStore: credentialFileStore,
+		NodeName:       os.Getenv("NODE_NAME"),
+		FileSystem:     filesystem.NewFileSystem(),
+		ResticExec:     restic.BackupExec{},
+		Log:            s.logger,
+
+		PvLister:  s.kubeInformerFactory.Core().V1().PersistentVolumes().Lister(),
+		PvcLister: s.kubeInformerFactory.Core().V1().PersistentVolumeClaims().Lister(),
+		PvbLister: s.veleroInformerFactory.Velero().V1().PodVolumeBackups().Lister(),
```

Comment:
Pod Volume Backup CRD is moved to controller-time. Maybe a seperated Lister is not needed. 
The Client should've been enable to get PVB.

Reply:
@blackpiglet, I agree that I should be able to get PodVolumeBackupList. I tried to get the list by updating the following in pkg/controller/pod_volume_backup_controller.go.

```go
func (r *PodVolumeBackupReconciler) getParentSnapshot(pvcUID, backupStorageLocation string) string {
	r.Log = r.Log.WithField("pvcUID", pvcUID)
	r.Log.Infof("Looking for most recent completed pod volume backup for this PVC")

	var pvbList velerov1api.PodVolumeBackupList
	err := r.Client.Get(r.Ctx, client.ObjectKey{}, &pvbList)
      
       [...clipped...]
```

Unfortunately, I encounter a compile error,

```go
cannot use &pvbList (value of type *"github.com/vmware-tanzu/velero/pkg/apis/velero/v1".PodVolumeBackupList) as client.Object value in argument to r.Client.Get: missing method GetAnnotations
```

I thought that the controller tools that Kubebuilder uses was supposed to ensure the PodVolumeBackupList type satisfied the client.Object interface. I verified in pkg/apis/velero/v1/pod_volume_backup_types.go that the correct marker, `//+kubebuilder:object:root=true` was set. 

I also ran `make update` in Velero project root to ensure all the controller tools were run again and new CRDs generated.

Is there something I'm missing? Why can't I use client.Client to get the PodVolumeBackupList without getting a compile error?

@zubron, have you encountered this error before?

Reply:
Currently, we have the following kubebuilder marker on the PodVolumeBackupList type:

```go
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Based on the error wanting client.Ojbect, I replaced it so the client.Object type would be generated:

```go
// +k8s:deepcopy-gen:interfaces=sigs.k8s.io/controller-runtime/pkg/client.Object
```

This also didn't work. I got this error:

```text
cannot use &pvbList (value of type *"github.com/vmware-tanzu/velero/pkg/apis/velero/v1".PodVolumeBackupList) as "sigs.k8s.io/controller-runtime/pkg/client".Object value in argument to r.Client.Get: wrong type for method DeepCopyObject (have func() sigs.k8s.io/controller-runtime/pkg/client.Object, want func() k8s.io/apimachinery/pkg/runtime.Object)
```

@blackpiglet, you said,

> Pod Volume Backup CRD is moved to controller-time. 

However, I don't think PodVolumeBackupList type was moved to controller runtime, so I think we still need to use the old informer.

Reply:
@codegold79 Sorry I missed your mention here!

I don't know if this is the issue, but could you check if using the plural name (`podvolumebackups`) rather than the singular name (`podvolumebackup`) in the RBAC definitions fixes this issue? https://github.com/vmware-tanzu/velero/pull/4436/files#diff-72e6196edabb1cf7d4c410c8fad08976be6213dee85ad249a4ac3e856e440432R134-R135

I'm wondering if the list kind is associated with the plural name in some way :confused: I noticed in the existing controller runtime CRDs, we are using the plural name. https://github.com/vmware-tanzu/velero/blob/175d2967bfe630cc7ad1f1973cb4d8a293867efd/pkg/apis/velero/v1/download_request_types.go#L108-L109

Reply:
Thanks @zubron. I added [a commit](https://github.com/vmware-tanzu/velero/pull/4436/commits/52fabfd03a3705a52fb33f946462376f44c84a37) with the changes you suggested. I ran `make update` but it didn't seem to change any generated files or CRDs. Unless I missed running something, I have a feeling singular and plural work the same.

Reply:
@codegold79 Sorry to suggest that! I'm pretty sure you're right, that the singular and plural should work the same.

I just took another look at this and realised that you probably need to use `List` rather than `Get` for fetching the list of PVBs as the list type has [`ListMeta`](https://github.com/vmware-tanzu/velero/pull/4436/files#diff-72e6196edabb1cf7d4c410c8fad08976be6213dee85ad249a4ac3e856e440432R142) rather than `ObjectMeta`. I'm not quite sure how all the plumbing for this works, but I tried using that client to call List rather than Get and it compiled. Maybe that's enough? :woman_shrugging: This stuff is pretty complicated to follow!

Reply:
@zubron, you are a genius! Yes, you're absolutely right about using `r.Client.List` and not `r.Client.Get`. I tried it and so far it works. Thank you so much for looking into this and don't be sorry about the plural/singular thing. It just shows how sharp your eye is for detail. Anyway, I think I'll change it to plural to be consistent with the other controllers.

Ok, one problem down, a couple more to go. I'll work with Dave on those issues (status not updating, credentials file not being found) tomorrow. I'll let you know how it goes.

Reply:
Woohoo! So glad that worked 😄 Good luck with the next steps!

Reply:
@blackpiglet, with @zubron's help, I was able to make your suggestion to get rid of the old PVB lister work. Woot!